lock-api/src/main/java/com/palantir/lock/LockClient.java

```diff
@@ -66,6 +68,11 @@ public static LockClient of(String clientId) {
         Preconditions.checkArgument(!clientId.equals(INTERNAL_LOCK_GRANT_CLIENT_ID));
         return new LockClient(clientId);
     }
+    
+    @JsonCreator
+    public static LockClient valueOf(String clientId) {
```

Comment:
As a side note: this is not actually necessary for @PathParam because it has a single arg string constructor.  Yet, it is not really useful to use @PathParam in practice for us because dropwizard (as far as I know) does not handle the the serializing of the LockClient to a string (instead it uses Jersey's default of toString(), which is annoying).  I avoided dealing with this issue in https://github.com/palantir/atlasdb/pull/246 (I just kept things as strings), but if you know how to fix it to serialize independently of toString() feel free to also make that change here.